script.js

```diff
@@ -11,3 +45,313 @@ function hideColumnContents(className) {
+    }
+  }
+
+  columnlineThree.onmousedown = function(event) {
+    var oldColumnlineThreePosition = columnlineThree.offsetLeft;
+    var columnOneWidth = headerOne.offsetWidth;
+    var columnTwoWidth = headerTwo.offsetWidth;
+    var columnThreeWidth = headerThree.offsetWidth;
+    var columnFourWidth = headerFour.offsetWidth;
+    var oldmousePosition = event.clientX;
+
+    function moveToMouse(windowCoordinatesX) {
+      const newColumnlineLeft = windowCoordinatesX;
+      const newColumnlineLeftIsInMinWidth = newColumnlineLeft >= headerTwo.offsetLeft + minWidth && newColumnlineLeft <= headerThree.offsetLeft + headerThree.offsetWidth - minWidth;
+
+      if (newColumnlineLeftIsInMinWidth) {
+        columnlineThree.style.left = windowCoordinatesX + 'px';
+      }
+      else {
+        document.removeEventListener('mousemove', onMouseMove);
+        const diffColumnlineThree = oldColumnlineThreePosition - columnlineThree.offsetLeft;
+
+        let diffIsNegative = diffColumnlineThree < 0;
+
+        if (diffIsNegative) {
+          columnTwoWidth = columnlineThree.offsetLeft - headerOne.offsetWidth;
+          columnThreeWidth = minWidth;
+          adjustWidthOfColumns(columnTwo, columnThree, headerTwo, headerThree, columnTwoWidth, columnThreeWidth);
+        }
+  
+        if (!diffIsNegative) {
+          columnTwoWidth = minWidth;
+          columnThreeWidth = columnThreeWidth + diffColumnlineThree;
+          adjustWidthOfColumns(columnTwo, columnThree, headerTwo, headerThree, columnTwoWidth, columnThreeWidth);
+        }
+        
+        adjustWidthOfColumns(columnOne, columnFour, headerOne, headerFour, columnOneWidth, columnFourWidth);
+
+        return;
+      }
+    }
+
+    moveToMouse(event.clientX);
+
+    function onMouseMove(event) {
+      moveToMouse(event.clientX);
+    }
+
+    document.addEventListener('mousemove', onMouseMove);
+
+    columnlineThree.onmouseup = function(event) {
+      document.removeEventListener('mousemove', onMouseMove);
+      const diffMousePosition = oldmousePosition - event.clientX;
+
+      columnTwoWidth = headerTwo.offsetWidth;
+      columnThreeWidth = headerThree.offsetWidth;
+
+      const diffIsNegative = diffMousePosition < 0;
+
+      if (diffIsNegative) {
+        columnTwoWidth = columnTwoWidth - diffMousePosition;
+        columnThreeWidth = columnThreeWidth + diffMousePosition; 
+
+        adjustWidthOfColumns(columnTwo, columnThree, headerTwo, headerThree, columnTwoWidth, columnThreeWidth);
+      }
+
+      if (!diffIsNegative) {
+        columnTwoWidth = columnTwoWidth - diffMousePosition;
+        columnThreeWidth = columnThreeWidth + diffMousePosition;
+
+        adjustWidthOfColumns(columnTwo, columnThree, headerTwo, headerThree, columnTwoWidth, columnThreeWidth);
+      }
+
+      adjustWidthOfColumns(columnOne, columnFour, headerOne, headerFour, columnOneWidth, columnFourWidth);
+    }
+  }
+
+  columnlineFour.onmousedown = function(event) {
```

Comment:
Refactor this whole section. You should only declare the function used on "onmousedown" once.